src/include/gurt/dyn_hash.h

```diff
@@ -0,0 +1,347 @@
+/*
+ * (C) Copyright 2016-2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. 8F-30005.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/**
+ * \file
+ *
+ * Generic Dynamically Extended Hash Table APIs & data structures
+ */
+
+#ifndef __GURT_DYNHASH_H__
+#define __GURT_DYNHASH_H__
+
+#include <inttypes.h>
+#include <pthread.h>
+#include <stdbool.h>
+
+#include <gurt/list.h>
+#include <gurt/types.h>
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+typedef void *dh_item_t;
+struct d_hash_table;
+
+#define DYN_HASH_DEBUG 0
+
+/** @addtogroup GURT
+ * @{
+ */
+/******************************************************************************
+ * Generic Hash Table APIs / data structures
+ ******************************************************************************/
+struct dh_bucket;
+typedef struct dh_vector {
```

Comment:
(style)  do not add new typedefs

Reply:
Please use direct "struct dh_vector" instead of "dh_vector_t".